modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionServiceImpl.java

```diff
@@ -330,7 +330,10 @@ private AsyncCursor<List<Object>> executeExplain(ExplainPlan plan) {
     private void onMessage(String nodeName, QueryStartRequest msg) {
         assert nodeName != null && msg != null;
 
-        CompletableFuture<?> fut = sqlSchemaManager.actualSchemaAsync(msg.schemaVersion());
+        //TODO IGNITE-18733: We should use txTimestamp from message tx attributes to wait for metadata ready,
+        // then get actual version for txTimestamp and wait for catalog version ready.
+        // As optimization, we can lookup for latest available schema and go sync way if it's version >= desirable.
```

Comment:
in my opinion, this is not quite correct. It's better to use catalog version explicitly whenever possible

Reply:
Do you mean we shouldn't get rid of catalogVersion in message?
We already have an TxTimestmap that unambiguously maps to catalog version.